ipaplatform/redhat/tasks.py

```diff
@@ -180,68 +180,36 @@ def restore_pre_ipa_client_configuration(self, fstore, statestore,
                                              was_sssd_installed,
                                              was_sssd_configured):
 
-        auth_config = RedHatAuthConfig()
-        if statestore.has_state('authconfig'):
-            # disable only those configurations that we enabled during install
-            for conf in ('ldap', 'krb5', 'sssd', 'sssdauth', 'mkhomedir'):
-                cnf = statestore.restore_state('authconfig', conf)
-                # Do not disable sssd, as this can cause issues with its later
-                # uses. Remove it from statestore however, so that it becomes
-                # empty at the end of uninstall process.
-                if cnf and conf != 'sssd':
-                    auth_config.disable(conf)
-        else:
-            # There was no authconfig status store
-            # It means the code was upgraded after original install
-            # Fall back to old logic
-            auth_config.disable("ldap")
-            auth_config.disable("krb5")
-            if not(was_sssd_installed and was_sssd_configured):
-                # Only disable sssdauth. Disabling sssd would cause issues
-                # with its later uses.
-                auth_config.disable("sssdauth")
-            auth_config.disable("mkhomedir")
-
-        auth_config.execute()
+        auth_config = RedHatAuthTool.get_tool()
+        auth_config.unconfigure(
+            fstore, statestore, was_sssd_installed, was_sssd_configured
+        )
 
     def set_nisdomain(self, nisdomain):
-        # Let authconfig setup the permanent configuration
-        auth_config = RedHatAuthConfig()
-        auth_config.add_parameter("nisdomain", nisdomain)
-        auth_config.execute()
-
-    def modify_nsswitch_pam_stack(self, sssd, mkhomedir, statestore):
-        auth_config = RedHatAuthConfig()
-
-        if sssd:
-            statestore.backup_state('authconfig', 'sssd', True)
-            statestore.backup_state('authconfig', 'sssdauth', True)
-            auth_config.enable("sssd")
-            auth_config.enable("sssdauth")
-        else:
-            statestore.backup_state('authconfig', 'ldap', True)
-            auth_config.enable("ldap")
-            auth_config.enable("forcelegacy")
+        try:
+            with open(paths.SYSCONF_NETWORK, 'r') as f:
+                content = [
+                    line for line in f
+                    if not line.strip().upper().startswith('NISDOMAIN')
+                ]
+        except IOError:
+            content = []
```

Comment:
Why don't you use the same strategy with a call to get_tool() and implement a method specific to AuthConfig or AuthSelect?

Reply:
@flo-renaud, do you mean delegate nisdomain configuration to authselect class?

Reply:
@akokshar, yes